packages/core/src/routes/experience/classes/experience-interaction.ts

```diff
@@ -342,6 +356,34 @@ export default class ExperienceInteraction {
     );
   }
 
+  /**
+   * Verify the captcha token using current tenant's captcha provider.
+   *
+   * @param token The captcha token to verify.
+   *
+   * @throws {RequestError} with 422 if the captcha verification fails
+   */
+  public async verifyCaptcha(token: string) {
+    const captchaProvider = await this.tenant.queries.captchaProviders.findCaptchaProvider();
+
+    assertThat(captchaProvider, new RequestError({ code: 'session.captcha_failed', status: 422 }));
+
+    const captchaValidator = new CaptchaValidator(captchaProvider);
+    const isVerified = await captchaValidator.verifyCaptcha(token);
+
+    assertThat(isVerified, new RequestError({ code: 'session.captcha_failed', status: 422 }));
+
+    this.captchaVerified = true;
+  }
+
+  /**
+   * Skip the captcha verification for the current interaction,
+   * for social, sso, etc.
+   */
+  public skipCaptcha() {
+    this.captchaSkipped = true;
```

Comment:
Do we need to guard this method? E.g. only social and sso sign-in can skip captcha?

Reply:
We’ll ensure the condition is checked where the function is invoked.